decidim-core/lib/decidim/participatory_space_resourceable.rb

```diff
@@ -77,6 +77,7 @@ def link_participatory_spaces_resources(resources, link_name, data = {})
           end
         end
       end
+      alias_method :link_participatory_spaces_resources, :link_participatory_space_resources
```

Comment:
This is a little confusing for me. Why does this alias exists? It is to avoid changing all the calls to this method or because it has a semantical meaning?
If it is the first case, what do you think about replacing all (I found 8 matches in the entire project) and removing the alias to avoid the (very little 😄) unnecesary complexity?

Reply:
I added the alias to keep compatibility with any other community-developed module that might use this method, but I guess we could simply note it on the changelog.

I'll rename the current appearances of the old method name, remove the alias and update the changelog file to include a warning about the breaking change!

Thanks!

Reply:
Oh, I see. Now I'm thinking that it would be nice to introduce some kind of deprecations when is possible. In this case we could replace the `alias_method` with something like [this](http://seejohncode.com/2012/01/09/deprecating-methods-in-ruby/). What do you think? I don't know if it is too overkill for the current community size.

Reply:
Rails has a built-in system for deprecations, maybe I can use it: https://andycroll.com/ruby/use-a-deprecation-message/

Reply:
Done!

Reply:
Mmm, I would use both things, something like:

```ruby
module Deprecated
  # Define a deprecated alias for a method
  # @param [Symbol] name - name of method to define
  # @param [Symbol] replacement - name of method to (alias)
  def deprecated_alias(name, replacement)
    # Create a wrapped version
    define_method(name) do |*args, &block|
      ActiveSupport::Deprecation.warn "##{name} deprecated (please use ##{replacement})"
      send replacement, *args, &block
    end
  end
end
```

This is very easy to use, as you only need to replace the `alias_method` call with `deprecated_alias` to warn developers using it that they should change it, but without broking anything. Also, it helps to know that the alias exists, but it shouldn't be used.

Reply:
Ok, I'v epushed a new commit with a fix using your suggestion, @leio10 😄